compiler/rustc_mir/src/dataflow/move_paths/builder.rs

```diff
@@ -482,10 +481,8 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
             let base_ty = base_place.ty(self.builder.body, self.builder.tcx).ty;
             let len: u64 = match base_ty.kind() {
                 ty::Array(_, size) => {
-                    let length = size.eval_usize(self.builder.tcx, self.builder.param_env);
+                    let length: u64 = size.eval_usize(self.builder.tcx, self.builder.param_env);
                     length
```

Comment:
```suggestion
                    size.eval_usize(self.builder.tcx, self.builder.param_env);
```
we already have `let len: u64` above